src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,61 @@
+import { ReactPlugin, withAITracking } from '@microsoft/applicationinsights-react-js';
+import { ApplicationAnalytics, ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { ComponentType } from 'react';
+import ITelemetry from './ITelemetry';
+
+class Telemetry implements ITelemetry {
+  private appInsights: ApplicationInsights;
+  private config: any;
+
+  constructor() {
+    const areWeInDev = process.env.NODE_ENV === 'development';
+
+    this.config = {
+      instrumentationKey: process.env.REACT_APP_INSTRUMENTATION_KEY,
+      disableExceptionTracking: true,
+      disableTelemetry: areWeInDev ? false : true,
+    };
+
+    this.appInsights = new ApplicationInsights({
+      config: this.config
+    });
+  }
+
+  public initialize() {
+    this.appInsights.loadAppInsights();
+
+    this.appInsights.trackPageView();
+  }
+
+  public trackEvent(eventName: string, payload: any) {
+    if (!this.valid(eventName)) {
+      throw new Error('Invalid telemetry event name');
+    }
+
+    this.appInsights.trackEvent({ name: eventName }, payload);
+  }
+
+  public trackException(error: Error) {
+    const exception: any = { error: error.stack };
+    this.appInsights.trackException(exception);
+  }
+
+  public trackReactComponent(ComponentToTrack: ComponentType): ComponentType {
+    const reactPlugin = new ReactPlugin();
```

Comment:
I think this function will create multiple reactPlugin and appInsightsAnalytics plugins? AppInsights doesn't support adding a new plugin after initialized. If it is known that React component exist at the time AppInsights is initialized, it is recommend to add react plugin in `config` -> `extensions` on AI initialization, instruction adding react plugin: https://github.com/microsoft/ApplicationInsights-JS/tree/master/extensions/applicationinsights-react-js#basic-usage

Here's a react demo using AI react plugin, hope it helps: https://github.com/Azure-Samples/application-insights-react-demo/blob/000009395cccdf60b0c867fc0a26d995be3ab4a5/src/TelemetryService.js#L29